pkg/k8s/application/k8sApplicationService.go

```diff
@@ -674,6 +674,9 @@ func (impl *K8sApplicationServiceImpl) GetResourceList(ctx context.Context, toke
 		return resourceList, err
 	}
 	checkForResourceCallback := func(namespace, group, kind, resourceName string) bool {
+		if validateResourceAccess == nil { // if resource validate rbac func is nil then allow
+			return true
+		}
```

Comment:
I think , we should not allow it here, instead the caller should pass the function which will allow when called.

Reply:
did not do that because would need to manage separate func for superadmin role